### ISSUE TITLE
Enhancement: unbound-method is failing an unused property in object destructuring

### ISSUE THREAD

**Issue by bmarker** [typescript-eslint/typescript-eslint#12008]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/unbound-method/

### Description

The `unbound-method` rule flags destructuring a field only to be discarded.

If `typeof obj` includes `propertyToBeDiscarded` as an unbound method, this destructuring flags the unbound method:
```
const { propertyToBeDiscarded , ...rest } = obj
return {
  ...rest,
  newProperty,
}
```

You can avoid the `no-unused-vars` checks by setting a `_` prefix (or just `_`):
```
const { propertyToBeDiscarded: _ , ...rest } = obj
```

It would be nice if the `unbound-method` honored that too.

If not that, maybe a `ignoreRestSiblings` option similar to:
https://eslint.org/docs/latest/rules/no-unused-vars

Or `ignoreUnused` if it can tell it's unused.

### Fail

```TypeScript
interface IMyType {
  ...
  propertyToBeDiscarded(value: string): number
  ...
}

const obj: IMyType = { ... }
const { propertyToBeDiscarded: _ , ...rest } = obj
return {
  ...rest,
  newProperty,
}```

### Pass

```TypeScript
interface IMyType {
  ...
  propertyToBeDiscarded(value: string): number
  ...
}

const obj: IMyType = { ... }

// Passes but includes the unwanted `propertyToBeDiscarded` property unless you use `delete` or similar.
return {
  ...obj,
  newProperty,
}
```

### Additional Info

Ignoring the `_` prefix would be ideal for me. Implementing similar flags to `no-unused-vars` would also work.
https://eslint.org/docs/latest/rules/no-unused-vars

**Comment by Josh-Cena** [comment]
:/ I can acknowledge the use case, but idk a good way to solve it. Surely we cannot interrogate the options of another rule, and invoking scope analysis here seems unnecessary. Perhaps we can have an `ignore` option.

**Comment by bmarker** [comment]
I did not mean to use the actual setting on the other rule. Just the same name and perhaps be able to use the same logic about what cases to ignore.

**Comment by bradzacher** [comment]
I think this is just a "use a disable comment and move on" case.

I don't think we should try and add support for unused detection to the rule as that creates weird action at a distance behaviours to the rule.

I don't think we should add options to try and allow people to opt-out of this case because honestly 99.9999% of the time you only want to ignore a desteuctured member if and only if it is unused AND it has a rest sibling that is used.

I don't think we should add options to try and allow people to ignore specific name patterns because that's a recipe for creating horrible holes in your codebase.

So yeah - IMO this should be closed as a won't fix.

**Comment by JoshuaKGoldberg** [comment]
Thanks for filing, at least!

**Comment by bmarker** [comment]
We have inline configs disabled (too many exceptions slipped by when inline config was enabled). People seem better at catching and noticing if you actually have to modify the config to add an exception. Technically we could set an exception for the file.

The rest parameter would be sufficient to me. I forgot that the `_` was an option as well (`argsIgnorePattern`) and not just a global convention.

The alternative seems to be to enumerate all the fields you want instead of the fields you do not, which has its own problems.
